comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -53,14 +80,72 @@ type secretContext struct {
 	yamlPath string
 }
 
-func init() {
-	secretCache = make(map[string]string)
-	secretOrigin = make(handleToContext)
+// TODO: Hack to maintain a singleton reference to the secrets Component
+//
+// Only needed temporarily, since the secrets.Component is needed for the diagnose functionality.
+// It is very difficult right now to modify diagnose because it would require modifying many
+// function signatures, which would only increase future maintenance. Once diagnose is better
+// integrated with Components, we should be able to remove this hack.
+//
+// Other components should not copy this pattern, it is only meant to be used temporarily.
+var mu sync.Mutex
+var instance *secretResolver
+
+func newEnabledSecretResolver() *secretResolver {
+	return &secretResolver{
+		cache:   make(map[string]string),
+		origin:  make(handleToContext),
+		enabled: true,
+	}
 }
 
-func registerSecretOrigin(handle string, origin string, yamlPath []string) {
+func newSecretResolverProvider(deps dependencies) provides {
+	resolver := newEnabledSecretResolver()
+	resolver.enabled = deps.Params.Enabled
+
+	{
+		mu.Lock()
+		defer mu.Unlock()
+		if instance == nil {
+			instance = resolver
+		}
+	}
+
+	return provides{
+		Comp:          resolver,
+		FlareProvider: flaretypes.NewProvider(resolver.fillFlare),
+	}
+}
+
+func (r *secretResolver) IsEnabled() bool {
+	return r.enabled
+}
+
```

Comment:
Is this used anywhere ?

Reply:
Mentioned in another reply, removed now.